base/v0_5_exp/translate_test.go

```diff
@@ -1699,6 +1699,293 @@ func TestTranslateKernelArguments(t *testing.T) {
+	if err != nil {
+		t.Error(err)
+	}
+	tests := []struct {
+		name     string
+		in       PasswdUser
+		out      types.PasswdUser
+		report   string
+		fileDir  string
+		filePath string
+	}{
+		{
+			"empty user",
+			PasswdUser{},
+			types.PasswdUser{},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_inline",
+			PasswdUser{SSHAuthorizedKeys: []SSHAuthorizedKey{SSHAuthorizedKey(sshKey)}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local with multiple keys per file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyMultipleKeysFileName}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey), types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local and ssh_keys",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}, SSHAuthorizedKeys: []SSHAuthorizedKey{SSHAuthorizedKey(sshKey)}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey), types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local with multiple keys per file and ssh_keys",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyMultipleKeysFileName}, SSHAuthorizedKeys: []SSHAuthorizedKey{SSHAuthorizedKey(sshKey)}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey), types.SSHAuthorizedKey(sshKey), types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"non existing ssh_keys_local file name",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyNonExistingFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: open %FilePath%:",
+			sshKeyDir,
+			filepath.Join(sshKeyDir, "id_ed25519.pub"),
+		},
+		{
+			"empty ssh_keys_local file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyEmptyFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: local file to be embedded must not be empty\n",
+			sshKeyDir,
+			"",
+		},
+		{
+			"blank ssh_keys_local file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyBlankFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: local file to be embedded must not be empty\n",
+			sshKeyDir,
+			"",
+		},
+		{
+			"missing embed directory",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: local file paths are relative to a files directory that must be specified with -d/--files-dir\n",
```

Comment:
It's a bit cleaner to concatenate e.g. `ErrNoFilesDir.Error()` rather than hardcoding its message here.